### PR TITLE
Backports 0.16 pr5

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
builds and passes unit tests

### CODE REVIEW

**Comment by PastaPastaPasta** [comment]
0.14.0.4 (see how it includes reuse address box)

new page (doesn't have that box)

File: src/qt/intro.cpp [hunk 1]
```diff
@@ -44,7 +44,7 @@ class FreespaceChecker : public QObject
     Q_OBJECT
 
 public:
-    explicit FreespaceChecker(Intro *intro);
+    FreespaceChecker(Intro *intro);
```
Comment (UdjinM6) [inline src/qt/intro.cpp]: Why aef1eb0? Should be dropped imo.

**Comment by PastaPastaPasta** [comment]
fixed syntax error in receivedby.py, however that test still does not pass

File: test/functional/receivedby.py [hunk 2]
```diff
@@ -107,7 +107,7 @@ def run_test(self):
 
         # Create a new account named "mynewaccount" that has a 0 balance
         self.nodes[1].getaccountaddress("mynewaccount")
-        received_by_account_json = [r for r inself.nodes[1].listreceivedbyaccount(0, False, True)if r["account"] == "mynewaccount"][0]
+        received_by_account_json = [r for r in self.nodes[1].listreceivedbyaccount(0, True) if r["account"] == "mynewaccount"][0]
```
Comment (UdjinM6) [inline test/functional/receivedby.py]:
```suggestion
        received_by_account_json = [r for r in self.nodes[1].listreceivedbyaccount(0, False, True) if r["account"] == "mynewaccount"][0]
```

File: src/qt/intro.cpp (hunk 1 above)
Comment (PastaPastaPasta) [reply src/qt/intro.cpp]: because with out this commit it doesn't compile

Comment (PastaPastaPasta) [reply src/qt/intro.cpp]: maybe this fix is wrong but... idk

**Comment by UdjinM6** [comment]
Pls see https://github.com/UdjinM6/dash/commit/9aeaf39797147128db448a0d829205f6ce43ec28. The fix is a mix of few parts from https://github.com/bitcoin/bitcoin/pull/11824 (some other parts of which were already backported earlier via https://github.com/dashpay/dash/pull/2764 and https://github.com/dashpay/dash/pull/3025 btw) and few parts from https://github.com/bitcoin/bitcoin/pull/15855.

File: src/qt/intro.cpp (hunk 1 above)
Comment (UdjinM6) [reply src/qt/intro.cpp]: I had no issues compiling it with aef1eb0 reverted, see https://travis-ci.org/UdjinM6/dash/builds/640626624

**Comment by PastaPastaPasta** [comment]
cherry-picked.

File: src/qt/intro.cpp (hunk 1 above)
Comment (PastaPastaPasta) [reply src/qt/intro.cpp]: I'm not sure... maybe something with make caches, maybe `make clean` would have fixed it. Went ahead and reverted it

**Comment by UdjinM6** [comment]
LGTM

(Travis failure looks unrelated imo)

**Pull request merged**